linera-execution/src/wasm/system_api.rs

```diff
@@ -581,10 +584,28 @@ where
         caller: &mut Caller,
         operations: Vec<WriteOperation>,
     ) -> Result<(), RuntimeError> {
-        caller
-            .user_data_mut()
-            .runtime
-            .write_batch(Batch { operations })
+        WriteBatch::write_batch(&mut caller.user_data_mut().runtime, Batch { operations })
             .map_err(|error| RuntimeError::Custom(error.into()))
     }
 }
+
+// TODO(#1977): Remove once the WIT interface does not include `write-batch` in the service system
+// API
+/// An extension trait to separate the behavior between the contract runtime and the service
+/// runtime.
+pub trait WriteBatch {
```

Comment:
Why not having this function in `ContractRuntime` and (for now) `ServiceRuntime` ?

Reply:
That works on the host side, but makes it complicated inside the guest SDK. That's because there's only one `KeyValueStore` shared between the contract and the service, and we'd need to remove the `ViewSystemApi` first to clean that up :/